### ISSUE TITLE
-- Detecting libclang builtin headers directory -- fail

### ISSUE THREAD

**Issue by tachidok** [Sarcasm/irony-mode#414]
Ubuntu 16.04

When I run the command `irony-install-server` within emacs I get the following line in the mini-buffer
```
Install command: cmake -DCMAKE_INSTALL_PREFIX\=/home/tachidok/.emacs.d/irony/ /home/tachidok/.emacs.d/elpa/irony-20170725.1249/server && cmake --build . --use-stderr --config Release --target install
```

then after hitting RET I get the following error

```
-*- mode: compilation; default-directory: "/tmp/build-irony-server-1.0.0/" -*-
Compilation started at Thu Aug 10 18:30:28

cmake -DCMAKE_INSTALL_PREFIX\=/home/tachidok/.emacs.d/irony/ /home/tachidok/.emacs.d/elpa/irony-20170725.1249/server && cmake --build . --use-stderr --config Release --target install
-- Detecting libclang builtin headers directory
-- Detecting libclang builtin headers directory -- fail
CMake Warning at cmake/CheckLibClangBuiltinHeadersDir.cmake:82 (message):
  CheckLibClangBuiltinHeadersDir: failed to compile checker, please report.

    Compile output:
      Change Dir: /tmp/build-irony-server-1.0.0/CMakeFiles/CMakeTmp

  

  Run Build Command:"/usr/bin/make" "cmTC_19ad1/fast"

  /usr/bin/make -f CMakeFiles/cmTC_19ad1.dir/build.make
  CMakeFiles/cmTC_19ad1.dir/build

  make[1]: Entering directory
  '/tmp/build-irony-server-1.0.0/CMakeFiles/CMakeTmp'

  Building CXX object
  CMakeFiles/cmTC_19ad1.dir/LibClangDiagnosticsChecker.cpp.o

  /usr/bin/c++ -I/usr/lib/llvm-3.8/include -fno-exceptions -o
  CMakeFiles/cmTC_19ad1.dir/LibClangDiagnosticsChecker.cpp.o -c
  /tmp/build-irony-server-1.0.0/CMakeFiles/CMakeTmp/LibClangDiagnosticsChecker.cpp


  
  /tmp/build-irony-server-1.0.0/CMakeFiles/CMakeTmp/LibClangDiagnosticsChecker.cpp:9:27:
  fatal error: clang-c/Index.h: No such file or directory

  compilation terminated.

  CMakeFiles/cmTC_19ad1.dir/build.make:65: recipe for target
  'CMakeFiles/cmTC_19ad1.dir/LibClangDiagnosticsChecker.cpp.o' failed

  make[1]: *** [CMakeFiles/cmTC_19ad1.dir/LibClangDiagnosticsChecker.cpp.o]
  Error 1

  make[1]: Leaving directory
  '/tmp/build-irony-server-1.0.0/CMakeFiles/CMakeTmp'

  Makefile:126: recipe for target 'cmTC_19ad1/fast' failed

  make: *** [cmTC_19ad1/fast] Error 2

  

Call Stack (most recent call first):
  src/CMakeLists.txt:8 (check_libclang_builtin_headers_dir)


-- Irony package version is '1.0.0'
-- Found emacs: /usr/bin/emacs
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/build-irony-server-1.0.0
[ 14%] Building CXX object src/CMakeFiles/irony-server.dir/Command.cpp.o
In file included from /home/tachidok/.emacs.d/elpa/irony-20170725.1249/server/src/Command.h:14:0,
                 from /home/tachidok/.emacs.d/elpa/irony-20170725.1249/server/src/Command.cpp:11:
/home/tachidok/.emacs.d/elpa/irony-20170725.1249/server/src/support/CIndex.h:12:27: fatal error: clang-c/Index.h: No such file or directory
compilation terminated.
src/CMakeFiles/irony-server.dir/build.make:110: recipe for target 'src/CMakeFiles/irony-server.dir/Command.cpp.o' failed
make[2]: *** [src/CMakeFiles/irony-server.dir/Command.cpp.o] Error 1
CMakeFiles/Makefile2:981: recipe for target 'src/CMakeFiles/irony-server.dir/all' failed
make[1]: *** [src/CMakeFiles/irony-server.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2

Compilation exited abnormally with code 2 at Thu Aug 10 18:30:29

```

I installed clang-3.5, clang-3.5-dev, libclang-3.5-dev, llvm-3.5, llvm-3.5-dev using 'apt-get install'
in a linux terminal but I am having the same problem. I also tried `locate Index.h` but found
nothing. What is wrong?

I installed version 3.5 (based on #167) with the dev version as mentioned in #226

**Comment by tachidok** [comment]
Hi, I managed to have it working. I deleted all `clang` versions and reinstalled them using the default version given by Ubuntu (in my case it was version 3.8). I also installed `libclang-3.8-dev`.

Then I typed `M-x irony-mode` in emacs and ask me to run the `irony-install-server` which used these options

```
Install command: cmake -DCMAKE_INSTALL_PREFIX\=/home/tachidok/.emacs.d/irony/ /home/tachidok/.emacs.d/elpa/irony-20170725.1249/server && cmake --build . --use-stderr --config Release --target install
```
and **compiled as a charm**

NOTE: CMake users (as me). Remember to add the CMake flag `-DCMAKE_EXPORT_COMPILE_COMMANDS=1` **to your project** when you run cmake so that it generates the `compile_commands.json` file which I think is the one used by irony to perform the autocompletion. Check the **Compilation Database** section in this [link](https://github.com/Sarcasm/irony-mode).

Cheers
